org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/rename/RenameService2.xtend

```diff
@@ -95,13 +86,13 @@ class RenameService2 implements IRenameService2 {
 			val xtextResource = resourceSet.getResource(context.resource.URI, true)
 			if (xtextResource instanceof XtextResource) {
 				val position = options.renameParams.position
-				var EObject element;
+				var EObject element
 				try {
 					element = xtextResource.getElementAtOffset(context.document, position)
 				} catch (IndexOutOfBoundsException exc) {
 					issueAcceptor.add(FATAL, '''Invalid document �position.toPositionFragment(uri)�''')
 				}
-				if (issueAcceptor.maximumSeverity !== FATAL && (element === null || element.eIsProxy)) {
+				if (issueAcceptor.maximumSeverity !== FATAL && element === null || element.eIsProxy) {
```

Comment:
Doesn't that now throw an NPE when element is `null` and the severity is not FATAL?

Reply:
Funny, I could have asked the same; it's your code. So what was your intention? First, I change it because it looked odd, then I reverted.

Reply:
- If it's already fatal, don't add another error (it's likely a follow up)
- If not, and the element could not be detected (null) or is a unresolvable line (proxy, guarded by the former check), put accept the error
    Am I missing something?

Reply:
Also, why to check `issueAcceptor.maximumSeverity !== FATAL` at all? If it has `ERROR` or `FATAL`, it already failed so it does not matter. Otherwise, our intention is to fail here. Thanks!

Reply:
I see. It may have been a bad idea to throw an exception in the issueAcceptor on ERRORS and WARNINGS. Clients should not have to rely on such assumptions. Feel free to refactor.

The entire thing was inspired by Eclipse LTK, where you can collect all kinds of errors and warnings on refactoring and report them to the user. LSP does not provide any proper error channel, and the entire thing got mixed up in order to have a simpler control flow.

Reply:
OK, I will change the issue acceptor.